src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -204,7 +207,12 @@
    agent-shutdown-fn :- IFn
    profiler :- (schema/maybe PuppetProfiler)
    metrics-service]
-  (let [initialize-pool-instance-fn (get-initialize-pool-instance-fn jruby-puppet-config profiler metrics-service)
+  (let [modified-jruby-puppet-config (if (get-in jruby-config [:multithreaded])
```

Comment:
I think you can just use `(:multithreaded jruby-config)` here, since you're not looking for a nested key.